python/pypto/runtime/runner.py

```
@@ -490,6 +499,160 @@ def from_run_config(cls, cfg: "RunConfig") -> "_DfxOpts":
+    swimlane pass registers its own (see :func:`_execute_dfx_passes`). The spec
+    tells :mod:`pypto.runtime._dep_gen_capture` how to rebuild the orch args.
+
+    Failure is logged, not raised: the swimlane pass still runs, just without a
+    captured graph (lanes degrade to anonymous ``task(rXtY)`` with no arrows).
+    """
+    dfx_dir.mkdir(parents=True, exist_ok=True)
+    spec_path = dfx_dir / "_dep_gen_spec.json"
+    spec_path.write_text(json.dumps(spec), encoding="utf-8")
+    cmd = [sys.executable, "-m", "pypto.runtime._dep_gen_capture", str(spec_path)]
+    try:
+        subprocess.run(cmd, check=True)
```

Ruff (0.15.17)

[error] `subprocess` call: check for execution of untrusted input

(S603)